java/com/google/cose/SignMessage.java

```diff
@@ -0,0 +1,116 @@
+package com.google.cose;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.builder.ArrayBuilder;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.Map;
+import com.google.cose.exceptions.CoseException;
+import com.google.cose.utils.CborUtils;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Implements COSE_Sign class.
+ */
+public class SignMessage extends CoseMessage {
+  private final byte[] message;
+  private final List<Signature> signatures;
+
+  SignMessage(byte[] protectedHeaderBytes, Map unprotectedHeaders, byte[] message,
+      List<Signature> signatures) {
+    super(protectedHeaderBytes, unprotectedHeaders);
+    this.message = message;
+    this.signatures = signatures;
+  }
+
+  static class Builder {
+    private byte[] protectedHeaderBytes;
+    private Map unprotectedHeaders;
+    private byte[] message;
+    private List<Signature> signatures;
+
+    public SignMessage build() {
+      if ((protectedHeaderBytes != null) && (unprotectedHeaders != null) && (message != null)
+          && (signatures != null)) {
+        return new SignMessage(protectedHeaderBytes, unprotectedHeaders, message, signatures);
+      } else {
+        throw new CoseException("Some fields are missing.");
+      }
+    }
+
+    public Builder withProtectedHeaderBytes(byte[] protectedHeaderBytes) {
+      this.protectedHeaderBytes = protectedHeaderBytes;
+      return this;
+    }
+
+    public Builder withUnprotectedHeaders(Map unprotectedHeaders) {
+      this.unprotectedHeaders = unprotectedHeaders;
+      return this;
+    }
+
+    public Builder withMessage(byte[] message) {
+      this.message = message;
+      return this;
+    }
+
+    public Builder withSignatures(List<Signature> signatures) {
```

Comment:
I think a more idiomatic way of doing this would be to have an `addSignature(Signature signature)` and each time it's called it extends the list with the new signature.

Reply:
or withSignatures(Signature...signatures) which can take multiple parameters and just extends the signature list.